src/cpp/cpp_typecheck_expr.cpp

```diff
@@ -121,7 +121,6 @@ void cpp_typecheckt::typecheck_expr_main(exprt &expr)
     // these appear to have type "struct _GUID"
     // and they are lvalues!
     expr.type() = struct_tag_typet("tag-_GUID");
-    follow(expr.type());
```

Comment:
Was this always redundant?

Reply:
It was introduced in this way in 0b2c6d87 and even back then `follow` didn't have side effects, so I'd say: yes.